lib/model-api/utils.rb

```diff
@@ -419,10 +419,49 @@ def process_collection_includes(collection, opts = {})
           includes << attr_metadata[:key] if attr_metadata[:type] == :association
         end
         includes = includes.compact.uniq
+        includes = remove_excluded_associations(includes, opts)
         collection = collection.includes(includes) if includes.present?
         collection
       end
 
+      # Does not eager load associations mentioned in the exclude_assocations
+      # array defined by user.
+      def remove_excluded_associations(includes, opts)
+        includes_dup = includes.compact.uniq.deep_dup
+        exclude_associations = opts[:exclude_associations]
+        return includes_dup unless exclude_associations.present?
+        exclude_associations.each do |association|
+          remove_association_from_array(includes_dup, association.to_sym)
+        end
+        includes_dup
+      end
+
+      # If the association is present in array form (in collection_includes) delete it.
+      # Then loop over the array elements to remove the association incase of nested associations.
+      def remove_association_from_array(array_collection, association)
+        array_collection.delete(association)
+        array_collection.each_with_index do |collection, i|
+          if collection.is_a?(Hash)
+            array_collection[i] = remove_association_from_hash(collection, association)
+          end
+        end
+        array_collection
+      end
+
+      # Incase of nested associations (in collection_inlcudes) delete the key with the association name
+      # and then loop over to delete associations present as element of a array if any
```

Comment:
`loop over to delete associations present as element of a array if any`
seems little confusing,
`loop over the collection to delete associations present as element of the array if any` helps?